lib/KalturaFTPServer.js

```diff
@@ -176,34 +192,41 @@ FtpSession.prototype = {
 			}
 		}
 	},
-	handleICalListResponse: function (listResponse, callback, currentList, This, service, pager, currentClient, handler, filter) {
-		var eventsRegex = /BEGIN:VEVENT([\s\S]*?)END:VEVENT/g;
-		var results = listResponse.match(eventsRegex);
-		if (results == null) {
-			callback(null, currentList);
+	handleICalListResponse: function (listResponse, callback, currentList, This, service, pager, currentClient, handler, filter, retrieveOnePageOnly) {
+		var eventsRegex = /BEGIN:VERROR([\s\S]*?)END:VERROR/g;
+		var eventError = listResponse.match(eventsRegex);
+		if (eventError != null) {
+			This.logErrorMsg(eventError);
+			callback(eventError, currentList);
 		} else {
-			for (var i = 0; i < results.length; i++) {
-				var event = results[i];
-				var eventIdRegex = /(?:^|\n)x-kaltura-id:([^\s]+)/i;
-				var eventId = event.match(eventIdRegex).slice(1);
-
-				This.lists[service].push(event);
-				currentList.push(eventId + '.' + iCalExtention);
-				var eventId = eventId + '.' + iCalExtention;
-				This.logMsg('Adding item to cache item key ' + eventId);
-				parsedList[eventId] = event;
-			}
-			if (results.length == pager.pageSize) {
-				pager.pageIndex++;
-				currentClient[service].listAction(handler, filter, pager);
-			}
-			else {
+			var eventsRegex = /BEGIN:VEVENT([\s\S]*?)END:VEVENT/g;
+			var results = listResponse.match(eventsRegex);
+			if (results != null) {
+				for (var i = 0; i < results.length; i++) {
+					var event = results[i];
+					var eventIdRegex = /(?:^|\n)X-KALTURA-ID:([^\s]+)/i;
+					var eventId = event.match(eventIdRegex).slice(1);
+					event = 'BEGIN:VCALENDAR' + os.EOL + event + os.EOL + 'END:VCALENDAR';
```

Comment:
Don't use os.EOL, the iCal standard demands that it will be always CRLF, see https://tools.ietf.org/html/rfc5545#section-3.1.